src/core/core.datasetController.js

```diff
@@ -236,6 +238,33 @@ helpers.extend(DatasetController.prototype, {
 		me.resyncElements();
 	},
 
+	/**
+	 * Returns the merged user-supplied and default dataset-level options
+	 * @private
+	 */
+	_configure: function() {
+		var me = this;
+		var dataset = me.getDataset();
+		var chartDefaults = me.chart.options.datasets;
+		var datasetDefaults = helpers.merge({}, [defaults.global.datasets[me._type], chartDefaults ? chartDefaults[me._type] : {}]);
```

Comment:
I don't think we need to merge anything here, but simply use `me.chart.options.datasets[me._type]`.